### PR TITLE
Cleanups - rebased on 4.9.1-rc.1

### PR DESCRIPTION
**Pull request by stackchain**

### CODE REVIEW

**Comment by stackchain** [comment]
#2271

File: src/components/Boundary/Boundary.tsx [hunk 1]
```diff
@@ -22,52 +22,50 @@ import LocalizableError from '../../i18n/LocalizableError'
...
+  enabled?: boolean
 }
-export const LoadingBoundary = ({children, ...props}: LoadingBoundaryProps) => {
-  if (props.loading?.enabled === false) return <>{children}</>
+export const LoadingBoundary = ({children, ...props}: LoadingBoundaryProps & {children: React.ReactNode}) => {
+  if (props.enabled === false) return <>{children}</>
 
   return (
-    <React.Suspense fallback={props.loading?.fallback ?? <LoadingFallback {...props.loading?.fallbackProps} />}>
+    <React.Suspense fallback={props.fallback === undefined ? <LoadingFallback {...props.fallbackProps} /> : null}>
```
Comment (michalszorad) [inline src/components/Boundary/Boundary.tsx]: Was it intentional to check for `undefined`? This way `<LoadingBoundary fallback={<Something />} />` won't work

File: src/TxHistory/useOnScroll.ts [hunk 2]
```diff
@@ -2,36 +2,39 @@ import {useRef} from 'react'
...
+    const topReached = nativeEvent.contentOffset.y === 0
+    const scrollOffset = nativeEvent.contentOffset.y
 
     if (topBounce) return // ignore top bounce
     if (bottomBounce) return // ignore bottom bounce
 
-    const scrollUp = nativeEvent.contentOffset.y < scrollOffset.current
-    const scrollDown = nativeEvent.contentOffset.y > scrollOffset.current
+    const scrollUp = scrollOffset < previousScrollOffset.current - buffer
+    const scrollDown = scrollOffset > previousScrollOffset.current + buffer
```
Comment (michalszorad) [inline src/TxHistory/useOnScroll.ts:26]: I've tested this on an emulator, and the header does not collapse when I scroll slowly. Is this expected?

File: src/components/Boundary/Boundary.tsx (hunk 1 above)
Comment (wolverineks) [reply src/components/Boundary/Boundary.tsx]: there are a bunch of comments on the previous version
https://github.com/Emurgo/yoroi-mobile/pull/2271

Comment (wolverineks) [reply src/components/Boundary/Boundary.tsx]: maybe `!"fallback" in props` is more readable?

File: src/TxHistory/useOnScroll.ts (hunk 2 above)
Comment (wolverineks) [reply src/TxHistory/useOnScroll.ts:26]: not sure how to write this better,
slow scroll without buffer has jitters

**Review by michalszorad — APPROVED** [review]

**Review by banklesss — APPROVED** [review]

**Pull request merged**